mesonbuild/compilers/swift.py

```diff
@@ -115,6 +117,29 @@ def get_include_args(self, path: str, is_system: bool) -> T.List[str]:
     def get_compile_only_args(self) -> T.List[str]:
         return ['-c']
 
+    def get_options(self) -> MutableKeyedOptionDictType:
+        opts = super().get_options()
+
+        key = self.form_compileropt_key('lang_version')
+        opts[key] = options.UserStringOption(
+            self.make_option_name(key),
+            'Swift language version.',
+            '')
```

Comment:
These are generally combo options where we set the values we know and use `"none"` as the sentinel value.

I have a TODO to extend this to allow other values and do a compiler check them by running the compiler, but I'd prefer to keep with the "we have a list of known good values" for consistency

Reply:
Ah, I couldn't figure out how to get swiftc to print the valid values for it at first and I didn't find a list of them anywhere either so I left it like this. Fixed!